app/service/url_shortener.py

```diff
@@ -1,33 +1,52 @@
+import uuid
 
 from pydantic import BaseModel
 from typing import Optional
-
+from app.models.database import LinkDatabase
+from app.models.link import Link
 
 """
 This file is a helper that recieves data from api_handlers 
 and converts it into the short_link format needed for the databse
 """
 
+
+
 #link class defines how the data should look.
-class Link(BaseModel):
-    id: Optional[str]
-    original_url: str
-    short_url: str
-
-def create_shortened_url(original_url: str) -> str:
-    pass
-    """ This function will take information from api_handlers, call generate_url, create a link object  
-        send the information to database.py for insertion
-        returns the generated valid shortlink
-    """
 
 
-def generate_url() -> str:
-    pass
-    """
-    This function is a helper function of create_shortened url, and will handle the generation of a link
-    and ensuring it doesnt already exist
-    """
+def create_shortened_url(original_url: str, short_url: Optional[str], username: str) -> str:
+
+    # open a connection to the DB
+    db = LinkDatabase("URLShortenerTable")
+
+    # if no shortlink was provided, create one
+    if not short_url:
+        short_url = generate_url(db)  # Generate short URL if not provided
+
+    # create a Link object
+    link = Link(original_url=str(original_url), short_url=short_url)
+
+    # create an entry in the DB
+    if db.create_link(link, username):
+        return link.short_url
+
+    # if something went wrong
+    else:
+        raise Exception("Database insertion failed")
+
+def generate_url(db) -> str:
+
+    #create a 8 digit uuid as shortlink
+    url = str(uuid.uuid4())[:8]
```

Comment:
This should be refactored to a function called `generate_random_short_url`. This can avoid duplicate code.

Reply:
This already is a helper function?